tests/ren-integration/adapter_test.js

```diff
@@ -0,0 +1,302 @@
+  }
+
+  // Approve eth account for spending of wbtc/renbtc.
+  const wbtc = new web3.eth.Contract(WBTCFaucetABI, KOVAN_WBTC_TOKEN_ADDR);
+  await approveSpend(wbtc);
+  const renBTC = new web3.eth.Contract(renBTCABI, KOVAN_RENBTC_TOKEN_ADDR);
+  await approveSpend(renBTC);
+});
+
+describe('BadgerRenAdapter', function() {
+  this.timeout(60 * MINUTES); // 60 minute t/o for integration tests
+
+  it('should mint renBTC', async () => {
+    const params = {
+      asset: 'BTC',
+      from: Bitcoin(),
+      to: Ethereum(web3.currentProvider, ETHEREUM_NETWORK).Contract({
+      sendTo: KOVAN_ADAPTER_ADDR,
+      contractFn: 'mintRenBTC',
+      // Arguments expected for calling `mint`
+      contractParams: [
+        {
+          name: '_to',
+          type: 'address',
+          value: web3.eth.defaultAccount,
+        },
+      ],
+    }),
+  };
+
+    const mint = await renJS.lockAndMint(params);
+
+    logger.info('processing renBTC mint...');
+    const amount = 0.00101;
+    await processMint(mint, amount);
+  });
+
+  it('should mint wBTC', async () => {
+    const maxSlippage = 0.10;
+    const params = {
+      asset: 'BTC',
+      from: Bitcoin(),
+      to: Ethereum(web3.currentProvider, ETHEREUM_NETWORK).Contract({
+        sendTo: KOVAN_ADAPTER_ADDR,
+        contractFn: 'mintWBTC',
+        // Arguments expected for calling `mint`
+        contractParams: [
+          {
+            name: '_slippage',
+            type: 'uint256',
+            value: Number(maxSlippage * 10000).toFixed(0),
+          },
+          {
+            name: '_to',
+            type: 'address',
+            value: web3.eth.defaultAccount,
+          },
+        ],
+      }),
+    };
+
+    const mint = await renJS.lockAndMint(params);
+
+    logger.info('processing wBTC mint...');
+    const amount = 0.00101;
+    await processMint(mint, amount);
+  });
+
+
+  it('should burn renBTC', async () => {
+    const recipient = await account.address('btc');
+    const amount = 0.00101;
+    const params = {
+      // Send BTC from Ethereum back to the Bitcoin blockchain.
+      asset: 'BTC',
+      to: Bitcoin().Address(recipient),
+      from: Ethereum(web3.currentProvider).Contract((btcAddress) => ({
+        sendTo: KOVAN_ADAPTER_ADDR,
+        contractFn: 'burnRenBTC',
+        contractParams: [
+            {
+                type: 'bytes',
+                name: '_to',
+                value: Buffer.from(btcAddress),
+            },
+            {
+                type: 'uint256',
+                name: '_amount',
+                value: RenJS.utils.toSmallestUnit(amount, 8),
+            },
+        ],
+        // NB: Need to include gas for burn tx as the gas cost of the burn method
+        // is not estimatable.
+        txConfig: { gas: 1000000 },
+      })),
+    };
+
+    const burn = await renJS.burnAndRelease(params);
+    logger.info('processing renBTC burn...');
+    await processBurn(burn);
+  });
+
```

Comment:
Should we consider adding scenarios:
1. Pool unbalanced
2. Negative test cases: arguments outside of boundary conditions to test assertions prevent errors